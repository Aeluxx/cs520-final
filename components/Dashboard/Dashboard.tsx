import { useState } from 'react'
import { Box, Grid } from '@mui/material'
import useAppBarHeight from '../../hooks/useAppBarHeight'
import ClassDrawer from './components/ClassDrawer'
import DocumentCard from './components/DocumentCard'
import SearchBar from './components/SearchBar'
import useAxios from 'axios-hooks'
import { DocumentType } from '../../types'
import CreateNewNoteButton from './components/CreateNewNoteButton'

export default function Dashboard() {
  const [selectedSectionId, setSelectedSectionId] = useState<string>('')
  const [{data: notes}] = useAxios<DocumentType[]>({url: 'http://localhost:3000/api/notesFromSection', method: 'GET', params: {sectionId: selectedSectionId}})
  const appBarHeight = useAppBarHeight()
  const [searchValue, setSearchValue] = useState('')

  return (
    <Box sx={{ backgroundColor: 'action.selected', height: `calc(100vh - ${appBarHeight}px)` }}>
      <ClassDrawer selectedSectionId={selectedSectionId} setSelectedSectionId={setSelectedSectionId} width={270} />
      {/* Requires margin to account for width of drawer */}
      <div style={{ display: 'flex', flexDirection: 'column', marginLeft: '270px' }}>
        <Box
          sx={{
            display: 'flex',
            m: 5,
            gap: 2,
          }}>
          <SearchBar style={{ flexGrow: 1 }} value={searchValue} onChange={e => setSearchValue(e.target.value)} />
          <CreateNewNoteButton sectionId={selectedSectionId}/>
        </Box>
        {/* For some reason adding a margin to the grid messes things up... using a box instead */}
        <Box mx={5}>
          <Grid spacing={2} container>
            {notes?.map((note, i) => {
              return (
                <Grid key={i} item xs={12} sm={6} md={4} lg={3}>
                  <DocumentCard document={note} />
                </Grid>
              )
            })}
          </Grid>
        </Box>
      </div>
    </Box>
  )
}
