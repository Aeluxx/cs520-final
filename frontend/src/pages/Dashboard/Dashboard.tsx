import { useState } from 'react'
import { Box, Button, Grid } from '@mui/material'
import useAppBarHeight from '../../hooks/useAppBarHeight'
import ClassDrawer from './components/ClassDrawer'
import DocumentCard from './components/DocumentCard'
import SearchBar from './components/SearchBar'
import { useConfirmDialog } from '../../components/ConfirmDialog'

export default function Dashboard() {
  const documents = [
    {
      title: 'Notes',
      id: '1',
      course: 'CS520',
      tags: ['Quiz 2', 'Linear Algebra'],
      lastUpdated: new Date(Date.now()),
      favorited: true,
    },
    {
      title: 'Notes',
      id: '2',
      course: 'CS520',
      tags: ['Quiz 2', 'Linear Algebra'],
      lastUpdated: new Date(Date.now()),
      favorited: true,
    },
    {
      title: 'Notes',
      id: '3',
      course: 'CS520',
      tags: ['Quiz 2', 'Linear Algebra'],
      lastUpdated: new Date(Date.now()),
      favorited: true,
    },
    {
      title: 'Notes',
      id: '4',
      course: 'CS520',
      tags: ['Quiz 2', 'Linear Algebra'],
      lastUpdated: new Date(Date.now()),
      favorited: false,
    },
    {
      title: 'Notes',
      id: '5',
      course: 'CS520',
      tags: ['Quiz 2', 'Linear Algebra'],
      lastUpdated: new Date(Date.now()),
      favorited: false,
    },
    {
      title: 'Notes',
      id: '6',
      course: 'CS520',
      tags: ['Quiz 2', 'Linear Algebra'],
      lastUpdated: new Date(Date.now()),
      favorited: true,
    },
  ]
  const appBarHeight = useAppBarHeight()
  const [searchValue, setSearchValue] = useState('')
  const handleNew = () => {

  }
  return (
    <Box style={{ backgroundColor: '#f3f3f3', height: `calc(100vh - ${appBarHeight}px)` }}>
      <ClassDrawer width={270} />
      {/* Requires margin to account for width of drawer */}
      <div style={{ display: 'flex', flexDirection: 'column', marginLeft: '270px' }}>
        <Box sx={{
          display: 'flex',
          m: 5,
          gap: 2
        }}>
        <SearchBar style={{flexGrow: 1}} value={searchValue} onChange={e => setSearchValue(e.target.value)} />
        <Button variant='contained' color='primary'>Create New</Button>
        </Box>
        {/* For some reason adding a margin to the grid messes things up... using a box instead */}
        <Box mx={5}>
          <Grid spacing={2} container>
            {documents.map(document => {
              return (
                <Grid item xs={12} sm={6} md={4} lg={3}>
                  <DocumentCard document={document} />
                </Grid>
              )
            })}
          </Grid>
        </Box>
      </div>
    </Box>
  )
}
