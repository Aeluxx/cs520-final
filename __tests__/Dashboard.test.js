import React from 'react';
import '@testing-library/jest-dom';
import Dashboard from '../components/Dashboard/Dashboard';
import ShallowRenderer from 'react-test-renderer/shallow';

let realUseContext;
let useContextMock;

beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
});

afterEach(() => {
    React.useContext = realUseContext;
});

it("Dashboard renders without error", () => {
    useContextMock.mockReturnValue("Test Value");
    const shallow = new ShallowRenderer();
    const dashboard = shallow.render(<Dashboard />);
});

it("Dashboard matches Snapshot", () => {
    useContextMock.mockReturnValue("Test Value");
    const shallow = new ShallowRenderer();
    const dashboard = shallow.render(<Dashboard />);
    const result = shallow.getRenderOutput();
    expect(result).toMatchSnapshot();
});