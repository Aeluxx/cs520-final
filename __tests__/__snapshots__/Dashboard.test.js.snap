// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard matches Snapshot 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "backgroundColor": "action.selected",
      "height": "calc(100vh - 56px)",
    }
  }
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Class Dashboard
    </title>
    <html
      lang="en"
    />
  </HelmetWrapper>
  <ClassDrawer
    selectedSectionId=""
    setSelectedSectionId={[Function]}
    width={270}
  />
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "marginLeft": "270px",
      }
    }
  >
    <ForwardRef(Box)
      sx={
        Object {
          "display": "flex",
          "gap": 2,
          "m": 5,
        }
      }
    >
      <SearchBar
        onChange={[Function]}
        style={
          Object {
            "flexGrow": 1,
          }
        }
        value=""
      />
      <CreateNewNoteButton
        sectionId=""
      />
    </ForwardRef(Box)>
    <ForwardRef(Box)
      mx={5}
    >
      <ForwardRef(Grid)
        container={true}
        spacing={2}
      />
    </ForwardRef(Box)>
  </div>
</ForwardRef(Box)>
`;
